subsys/bluetooth/host/id.c

```diff
@@ -1254,6 +1344,151 @@ void bt_id_del(struct bt_keys *keys)
 		bt_le_ext_adv_foreach(adv_unpause_enabled, NULL);
 	}
 }
+
+static int conflict_check_and_replace(uint8_t id, struct bt_keys *keys)
+{
+	/* For the given key check if it has conflicts with other keys in the Resolving List
+	 * (such keys have BT_KEYS_ID_ADDED state and BT_KEYS_ID_CONFLICT flag set). If it does, we
+	 * need to remove the conflicting key from the Resolving List and add the new key.
+	 *
+	 * If the key is not in the Resolving List, we can add the new key right away.
+	 *
+	 * If advertiser for the conflicting key is enabled, we cannot remove the key from the
+	 * Resolving List, so we return an error.
+	 */
```

Comment:
Should this be added as a function doxygen comment, so it is shown in documentation. Even if it is a static function.